lib/sender.js

```diff
@@ -128,25 +124,19 @@ Sender.prototype.postTransfers = function * () {
   let transfers = this.transfers
   // TODO Theoretically we'd need to keep track of the signed responses
   for (let transfer of transfers) {
-    let auth = transfer === transfers[0] ? {
-      user: this.source_username,
-      pass: this.source_password
-    } : undefined
-    let transferReq = yield request({
-      method: 'put',
-      auth: auth,
-      uri: transfer.id,
-      body: transfer,
-      json: true
-    })
-    if (transferReq.statusCode >= 400) {
-      throw new Error('Remote error: ' + transferReq.statusCode + ' ' +
-        JSON.stringify(transferReq.body))
+    let transferRes = yield request
+      .put(transfer.id)
+      .auth(this.source_username, this.source_password)
```

Comment:
Won't the `this.source_username, this.source_password` pair only be valid for the first transfer?